### PR TITLE
Update reducer 1 to implement 1

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no organization description. #26 
- Currently the return value of the reducer does not fit our needs. #27 
- Currently there are no dates in the contribution calendar. #28 
- Currently the uniqueness of a member is defined by name and platformId. #29 
- Currently there is no way to get the platform source types. #30 
- Currently there is no way to get a contribution share. #31

**What is the new behavior (if this is a feature change)?**
- Now there is a organisation description. #26 
- Now the return value of the reducer function is refined. #27 
- Now there are start and end date of a calendar year. #28 
- Now the uniqueness of a member is defined by his url. #29
- Now there is a way to get platform source data. #30 
- Now there is a way to get a contribution share. #31 

**Other information**:
- Ref: 🐍

### CODE REVIEW

File: src/reducer/database/models.ts [hunk 1]
```diff
@@ -707,7 +709,6 @@ class Statistic extends osm.models.StatisticSO implements IStatistic {
    * @returns {object} A object with from and to date.
    * @description Get the correct from and to date. The calculation
    *              is based of wether this statistic is the current year or not.
```
Comment (pinterid) [inline src/reducer/database/models.ts:711]: Change to: "is based on wether this statistic is the current year or not."

File: src/reducer/helper/profile.ts [hunk 2]
```diff
@@ -80,8 +92,16 @@ function mergedProfile() {
     });
 
     let repositories = organization.getRepositories() as IRepository[];
+    /* 
+       Set the organization as owner for each repository within
+       the organization. To render a organization as a member
+       a list of member keys to render by is defined.
+    */
```
Comment (kleberbaum) [inline src/reducer/helper/profile.ts:99]: Should be more like:
```
    /* 
     * Set the organization as owner for each repository within
     * the organization. To render a organization as a member
     * a list of member keys to render by is defined.
     */
```

File: src/reducer/database/helper/calendar.ts [hunk 3]
```diff
@@ -38,7 +40,6 @@ function generateCalendarStructure(startDate: string, endDate: string) {
 
   for (
     let m = moment(startDate);
-
```
[the remark this answers is not in the document]
Comment (schettn) [reply src/reducer/database/helper/calendar.ts:41]: This is the header of a for loop declaration. Therefore the newline is not valid.

File: src/reducer/helper/statistic.ts [hunk 4]
```diff
@@ -40,6 +53,10 @@ interface IStatistic extends models.Statistic {
     current: models.Streak;
     streaks: models.Streak[];
   };
+  /*
+    Calendar: A contribution calendar in day format. 
+  */
```
[the remark this answers is not in the document]
Comment (schettn) [reply src/reducer/helper/statistic.ts:58]: I will change it to the headline format 'cause it is needed there for describing a field of
this interface.

**Pull request merged**